include/particle_utility/position_distribution.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef _NESO_UTILITY_POSITION_DIST_H_
```

Comment:
Yes, these would be useful.  I can help implement some other ones also.  Also kinda useful would be to generalise so that they can be distributed along a chord or in a 2d plane embedded in the 3d space...etc

Reply:
I put this in NESO as NESO already has a dependency on boost (and NESO-Particles does not). 

I agree that they should be more accessible either by moving them to a more central place in NESO (for the moment?) or NP and accept the boost dependency?

Reply:
@cmacmackin @oparry-ukaea your thoughts please. Context: we want to build upon this file with more functionality, which may gather more dependencies (at the moment its boost). NESO-Particles is nice and light weight and we don't want it encumbered by bloaty deps. Some users will want the functionality but not the whole of NESO. Options:
 - Put this file in NESO-Particles clapped around ifdefs so that it gets ignored if a dep e.g. boost isn't found. (not an elegant solution and somewhat questionable regarding build reproducibility)
 - Keep this here and request users include it from NESO into their projects (danger we don't get updates if they are "copy pasting" it)
 - Put it in a third place, e.g. NESO-Particles-Extras which NESO git submodules? (repo profileration!)
 - Any thing else?

Reply:
I've spun this off into https://github.com/ExCALIBUR-NEPTUNE/NESO/issues/144

Reply:
Let's keep this code where it is for this PR...